lib/HdtDocument.cc

```diff
@@ -368,6 +370,171 @@ NAN_METHOD(HdtDocument::SearchTerms) {
     info[4]->IsObject() ? info[4].As<Object>() : info.This()));
 }
 
+/******** HdtDocument#_readHeader ********/
+
+class ReadHeaderWorker : public Nan::AsyncWorker {
+  HdtDocument* document;
+  // JavaScript function arguments
+  Persistent<Object> self;
+  // Callback return values
+  vector<string> subjects, predicates, objects;
+  uint32_t totalCount;
+
+public:
+  ReadHeaderWorker(HdtDocument* document, Nan::Callback* callback, Local<Object> self)
+    : Nan::AsyncWorker(callback), document(document), totalCount(0) {
+        SaveToPersistent("self", self);
```

Comment:
Do we still need the `self` thingy? Thought we got rid of that since we're using promises.

Reply:
They're still there in the c++ code (tbh not sure whether it's needed). I'd say to just keep it here for consistency with other c++ methods